app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -18,49 +18,81 @@ class GetUspsProofingResultsJob < ApplicationJob
 
   discard_on GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError
 
+  def enrollment_attributes(enrollment, enrollment_completed)
+    {
+      enrollment_code: enrollment.enrollment_code,
+      enrollment_id: enrollment.id,
+      minutes_since_last_status_check: enrollment.minutes_since_last_status_check,
+      minutes_since_last_status_update: enrollment.minutes_since_last_status_update,
+      minutes_to_completion: enrollment_completed ? enrollment.minutes_since_established : nil,
+    }
+  end
+
   def perform(_now)
     return true unless IdentityConfig.store.in_person_proofing_enabled
 
-    proofer = UspsInPersonProofing::Proofer.new
-
+    @enrollment_outcomes = {
+      enrollments_checked: 0,
+      enrollments_errored: 0,
+      enrollments_expired: 0,
+      enrollments_failed: 0,
+      enrollments_in_progress: 0,
+      enrollments_passed: 0,
+    }
     reprocess_delay_minutes = IdentityConfig.store.
       get_usps_proofing_results_job_reprocess_delay_minutes
-    InPersonEnrollment.needs_usps_status_check(
+    enrollments = InPersonEnrollment.needs_usps_status_check(
       ...reprocess_delay_minutes.minutes.ago,
-    ).each do |enrollment|
-      # Record and commit attempt to check enrollment status to database
-      enrollment.update(status_check_attempted_at: Time.zone.now)
+    )
+
+    analytics.idv_in_person_usps_proofing_results_job_started(
+      enrollments_count: enrollments.count,
+      reprocess_delay_minutes: reprocess_delay_minutes,
+    )
+
+    check_enrollments(enrollments)
+
+    analytics.idv_in_person_usps_proofing_results_job_completed(**enrollment_outcomes)
```

Comment:
Looking at the ticket's request of "when the job ended", I'm wondering what the goal was? If it was to have insight into the duration of the job, should we add that as an explicit property? Otherwise it may be difficult to correlate timing of the start/completed events.

Reply:
Yeah good point. I'll add that

Reply:
Added in 7a4920407048d42ae175e919ec784b8a39664527